zcash_primitives/src/transaction/components/transparent.rs

```diff
@@ -96,6 +96,26 @@ pub struct OutPoint {
 }
 
 impl OutPoint {
+    /// The "null" `OutPoint` in the Bitcoin sense. This has the txid set to
+    /// all-zeroes, and the output index set to `u32::MAX`.
+    ///
+    /// From `BaseOutPoint::IsNull()` in zcashd:
+    /// ```c++
+    ///     return (hash.IsNull() && n == (uint32_t) -1);
+    /// ```
+    ///
+    /// The null `OutPoint` cannot refer to an actually existing output in a
+    /// valid block, because the maximum block size prevents a transaction
+    /// from containing `2^32` transparent outputs, and because constructing
+    /// such a transaction would require finding a (SHA-256d or BLAKE2b-256)
+    /// preimage of the all-zeroes hash.
+    pub const NULL: Self = OutPoint {
+        hash: [0; 32],
+        n: u32::MAX,
+    };
```

Comment:
I'm suspicious of this value; it seems to me that wherever this is needed, an `Option<OutPoint>` should be used instead. Why does this need to be a public constant?

Reply:
This has been removed from #1423.